src/dd/dd.rs

```diff
@@ -0,0 +1,679 @@
+        match self.f {
+            OutFileType::Reg(ref mut f) => f.write(buf),
+            OutFileType::Stdout(ref mut s) => s.write(buf),
+        }
+    }
+}
+
+pub fn uumain(_args: Vec<String>) -> i32 {
+    let signal = chan_signal::notify(&[Signal::INT, Signal::USR1]);
+    let (s_done, r_done) = chan::sync(0);
+
+    let cf = parse_args();
+    let cf1 = Arc::new(Mutex::new(cf));
+    let cf2 = Arc::clone(&cf1);
+
+    let handle = thread::spawn(move || {
+        let (inp, out) = setup(Arc::clone(&cf1));
+        copy(inp, out, cf1, s_done);
+    });
+
+    loop {
+        chan_select! {
+            signal.recv() -> signal => {
+                display_stat(Arc::clone(&cf2));
+                println!("signal: {:?}", signal);
+                if let Some(Signal::INT) = signal {
+                    exit(1);
+                }
+            },
+            r_done.recv() => break
+        }
+    }
+    let _ = handle.join();
+    0
+}
+
+pub fn get_u(val: &str) -> Option<usize> {
+    if val.is_empty() || (val.len() == 1 && !val.chars().nth(0).unwrap().is_digit(10))
+        || val.chars().nth(0).unwrap() == '-'
+    {
+        return None;
+    }
+
+    let mut pref = val;
+    let mut size: usize = match val.chars().nth(val.len() - 1) {
+        Some('k') => {
+            pref = &val[..val.len() - 1];
+            1024
+        }
+        Some('b') => {
+            pref = &val[..val.len() - 1];
+            512
+        }
+        Some(m) if !m.is_digit(10) && m != 'k' && m != 'b' => {
+            return None;
+        }
+        _ => 1,
+    };
+
+    if pref.is_empty() || pref.chars().nth(pref.len() - 1).unwrap() == 'x'
+        || pref.chars().nth(0).unwrap() == 'x'
+    {
+        return None;
+    }
+    for m in pref.split('x') {
+        if let Ok(n) = m.parse::<usize>() {
+            size *= n;
+        } else {
+            return None;
+        }
+    }
+
+    Some(size)
+}
+
+pub fn get_i(val: &str) -> Option<isize> {
+    if val.is_empty() || (val.len() == 1 && !val.chars().nth(0).unwrap().is_digit(10)) {
+        return None;
+    }
+
+    let mut pref = val;
+    let mut size: isize = match val.chars().nth(val.len() - 1) {
+        Some('k') => {
+            pref = &val[..val.len() - 1];
+            1024
+        }
+        Some('b') => {
+            pref = &val[..val.len() - 1];
+            512
+        }
+        Some(m) if !m.is_digit(10) && m != 'k' && m != 'b' => {
+            return None;
+        }
+        _ => 1,
+    };
+
+    if pref.is_empty() || pref.chars().nth(pref.len() - 1).unwrap() == 'x'
+        || pref.chars().nth(0).unwrap() == 'x'
+    {
+        return None;
+    }
+    //    for m in pref.split('x').collect::<Vec<&str>>() {
+    for m in pref.split('x') {
+        if let Ok(n) = m.parse::<isize>() {
+            size *= n;
+        } else {
+            return None;
+        }
+    }
+
+    Some(size)
+}
+
+pub fn err<S: Into<String>>(s: S) {
```

Comment:
You should probably change this to:
```rust
pub fn err<S: Into<String>>(s: S) -> ! {
    // stuff
}
```

This will let you avoid adding `None` and stuff in cases like:
```rust
if x {
    Some(0)
} else {
    err("err");
    None
}
```

Reply:
Changed.